### DISCUSSION TITLE
Is support for dynamic mipmapped textures intended?

### DISCUSSION THREAD

**Discussion by capdj** [StereoKit/StereoKit#551] [Q&A]
I'm creating a dynamic mipmapped texture in my app using 
`new Tex(TexType.Image | TexType.Dynamic, TexFormat.Rgba32)`

On the Windows flat screen renderer, this gives me the following error:

`[SK warning] sk_gpu: Dynamic textures don't support mip-maps!`

...and the texture doesn't draw. Switching to no mipmaps gets things working as expected.

However, on Android (Quest 2), dynamic mipmapped textures *do* seem to work exactly as desired! So I guess the question is:

-Is enabling dynamic mipmapped textures within my app, but for Android only, a reasonable step for me to take?
-If so, is the intention for StereoKit to support dynamic mipmapped textures on all platforms eventually, so this would be a temporary hack? Or are there technical reasons why some platforms will never support that combination?

Thanks!

**Comment by maluoi** [comment]
Under the hood here, GL actually does a good job of making mip-maps automatically! So in the code I just let it do its thing here. D3D11 on the other hand, I recall having some problems with, enough so that I ended up generating mip-maps manually. Because of that, SK's current D3D11 implementation has a few more limitations than the GL one! In particular it's on CPU, so I disabled it for dynamic textures, and it doesn't play well with NPoT textures either.

This was mostly a time constraint thing, and not an intended limitation! This is something I can take a peek at in the near future, I've been kinda meaning to re-visit this bit anyhow :)
